tests/codegen/nullability-tests/build.gradle.kts

```diff
@@ -2,59 +2,47 @@
  * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
  * SPDX-License-Identifier: Apache-2.0
  */
+import aws.sdk.kotlin.gradle.codegen.smithyKotlinProjectionSrcDir
 import aws.sdk.kotlin.gradle.dsl.skipPublishing
-import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
-import software.amazon.smithy.gradle.tasks.SmithyBuild
-import software.amazon.smithy.gradle.tasks.Validate as SmithyValidate
 
 plugins {
-    kotlin("jvm")
-    alias(libs.plugins.smithy.gradle)
+    alias(libs.plugins.kotlin.jvm)
+    id("aws.sdk.kotlin.gradle.smithybuild")
 }
 
 skipPublishing()
 
+val codegen by configurations.getting
+dependencies {
+    codegen(project(":codegen:smithy-kotlin-codegen"))
+    codegen(libs.smithy.cli)
+    codegen(libs.smithy.model)
+}
+
+tasks.generateSmithyProjections {
+    smithyBuildConfigs.set(files("smithy-build.json"))
+}
+
 val optinAnnotations = listOf("kotlin.RequiresOptIn", "aws.smithy.kotlin.runtime.InternalApi")
 kotlin.sourceSets.all {
     optinAnnotations.forEach { languageSettings.optIn(it) }
 }
 
-val projectionDirs = listOf(
-    "smithyprojections/nullability-tests/client-mode/kotlin-codegen",
-    "smithyprojections/nullability-tests/client-careful-mode/kotlin-codegen",
+val projections = listOf(
+    "client-mode",
+    "client-careful-mode",
 )
 
 kotlin.sourceSets.getByName("main") {
-    projectionDirs.forEach {
-        kotlin.srcDir(layout.buildDirectory.dir(it))
+    projections.forEach { projectionName ->
+        kotlin.srcDir(smithyBuild.smithyKotlinProjectionSrcDir(projectionName))
     }
 }
-tasks.withType<KotlinCompile> {
-    // generated code has warnings unfortunately
-    kotlinOptions.allWarningsAsErrors = false
-}
-
-tasks["smithyBuildJar"].enabled = false
-
-val codegen by configurations.creating
-dependencies {
-    codegen(project(":codegen:smithy-kotlin-codegen"))
-    codegen(libs.smithy.cli)
-}
-
-val generateSdk = tasks.register<SmithyBuild>("generateSdk") {
-    group = "codegen"
-    classpath = codegen
-    inputs.file(projectDir.resolve("smithy-build.json"))
-    inputs.files(codegen)
-}
-
-tasks.named<SmithyValidate>("smithyValidate") {
-    classpath = codegen
-}
 
 tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
-    dependsOn(generateSdk)
+    dependsOn(tasks.generateSmithyProjections)
+    // generated code has warnings unfortunately
```

Comment:
nit: can link this issue as a FIXME: https://github.com/awslabs/aws-sdk-kotlin/issues/1169 (just converted now from the internal issue)